mvp-spec/operator-api.md

```diff
@@ -213,93 +213,34 @@ cat response-operatorO.json body-id-and-preferences.json | npx json --merge -e '
   "timestamp": 1639059692793,
   "signature": "message_signature_xyz1234",
   "body": {
-    "preferences": {},
-    "identifiers": []
-  }
-}
-```
-
-#### Signature
-
-Signature of the concatenation of:
-
-```
-sender + '\u2063' +
-receiver + '\u2063' +
-timestamp
-```
-
-### GET /v1/json/readOrGetNewId
-
-#### Request
-
-<!-- The query string below is generated with taking the request-publisherP.json file, removing body, and encoding it as query string:
-npx encode-query-string -nd `cat request-publisherP.json | npx json -e 'this.body = undefined' -o json-0`
--->
-
-```http
-GET /v1/json/readOrGetNewId?sender=publisherP.com&timestamp=1639057962145&signature=message_signature_xyz1234
-```
-
-#### Signature
-
-Signature of the concatenation of:
-
-```
-sender + '\u2063' +
-receiver + '\u2063' +
-timestamp 
-```
-
-#### Response
-
-Note: list of identifiers **cannot** be empty.
-
-<!-- Update this code block with:
-cat response-operatorO.json body-id-and-preferences.json | npx json --merge -e 'this.body.preferences = {}'
--->
-
-```json
-{
-  "sender": "operatorO.com",
-  "timestamp": 1639059692793,
-  "signature": "message_signature_xyz1234",
-  "body": {
-    "preferences": {},
-    "identifiers": [
-      {
-        "version": 1,
-        "type": "prebid_id",
-        "value": "7435313e-caee-4889-8ad7-0acd0114ae3c",
-        "source": {
-          "domain": "operator0.com",
-          "timestamp": 1639643110,
-          "signature": "prebid_id_signature_xyz12345"
-        }
+    "newIdentifier": {
```

Comment:
I would keep the same structure with a list of "identifiers" and avoiding "newIdentifier". If you want to express the creation of the resource, maybe we can use the standard of HTTP by returning a 201 status instead of a 200.

Reply:
After a second thought, I'm not so sure that this would be appropriate: 201 would mean that a resource was created, but in this case the ID is "generated" but it's not stored as a cookie. So I think it's not appropriate to consider this as a resource creation.
In fact, it would be probably more appropriate to return a 404 but I'm concerned about processing the response content with 404 HTTP code.

Again, if we consider this call is always made by "our own" JS library, then processing a 404 response is probably fine, and more RESTfull. (and the fetch API [supports it](https://stackoverflow.com/questions/39297345/fetch-resolves-even-if-404))

Reply:
Interesting perspective. The 404 doesn't follow the principle of least surprise, here, IMHO.

I am wondering if it is not due to the nature of the ID and the flow of data. I think that we have an implicit state in the ID. First, it is transient and then it can be stored. Maybe we should express this state in the resource. What do you think ?

Example:

```json
{
  "sender": "operatorO.com",
  "timestamp": 1639059692793,
  "signature": "message_signature_xyz1234",
  "body": {
    "identifiers": [
      {
        "version": 1,
        "state": "transient",
        "type": "prebid_id",
        "value": "560cead0-eed5-4d3f-a308-b818b4827979",
        "source": {
          "domain": "operator0.com",
          "timestamp": 1639643110,
          "signature": "prebid_id_signature_xyz12345"
        }
      }
    ]
  }
}
```
The downside of this approach is that we have an extra field that may not be necessary once the identifier is stored.

Reply:
Side note concerning your first thought: It can be also error-prone to have similar names around the same place of the payload. Indeed, it is not easy to catch while reading a documentation. It makes me think about some guidances from Clean Code: we should avoid long names for variables and functions with a delta of a few letters.